### PR TITLE
Minor memory enhancement of dropping previous requests

### PR DESCRIPTION
**Pull request by FrancisMurillo**
I like the autocomplete support of Tern with Emacs. When I use it with `company`, I seem to hang from time to time whenever it makes a request to tern, it could be my just my hardware but I'm trying to pinpoint why it is choking up from time to time.

This pull request simply ignores the previous request made from being processed or triggered. This slightly helps when I am typing faster than it completes so it prevents the previous completions from triggering.

### CODE REVIEW

File: emacs/tern.el [hunk 1]
```diff
@@ -293,15 +293,17 @@ list of strings, giving the binary name and arguments.")
...
-                            (when type
-                              (setf tern-last-argument-hints (cons opening-paren type))
-                              (tern-show-argument-hints))))
-                        `((type . "type")
-                          (preferFunction . t))
-                        opening-paren
-                        :silent)))))
+        (lexical-let ((generation tern-command-generation))
+          (tern-run-query (lambda (data)
+                            (when (= tern-command-generation (1- generation))
```
Comment (marijnh) [inline emacs/tern.el]: Isn't `tern-run-query` already doing this test?

Comment (FrancisMurillo) [inline emacs/tern.el]: My memory fails me on this. 
I remember I was being aggressive in denying previous requests from executing.
Testing out the code again, you might be right.

Comment (marijnh) [inline emacs/tern.el]: Could you adjust the PR to revert this part?

**Comment by FrancisMurillo** [comment]
Removed redundant aggressive conditional

**Comment by marijnh** [comment]
Thanks. Merged as 391bb28

**Pull request closed without merging**